lib/modulesync/repository.rb

```diff
@@ -120,13 +120,10 @@ def submit_changes(files, options)
           repo.push('origin', branch, opts_push)
         end
       rescue Git::GitExecuteError => e
-        if e.message.match?(/working (directory|tree) clean/)
-          puts "There were no changes in '#{@directory}'. Not committing."
-          return false
-        else
-          puts e
-          raise
-        end
+        raise unless e.message.match?(/working (directory|tree) clean/)
```

Comment:
I agree this rescue block is a bit weird. Does it make sense to go a step further and prior to committing check if the working tree is dirty? I think this would work
```ruby
unless repo.status.added.any?
  puts "There were no changes in '#{@directory}'. Not committing."
  return false
end

Reply:
IMHO, this is the whole method `#submit_changes` that should be reworked.
It seems to do many things and some looks useless or weird to me (some examples):
* why amending a commit in this `msync` context?
* should `force` option be the default in `msync` context?
* the `pre_commit_hook` feature is too specific (as said in #127)
* the `remote_branch` handling seems overkill
* finally, the git exception handling...

To address these topics, I think its safer to wrote tests first because ATM, there are no behavior tests we can rely on to improve these points.

So if you are OK, we keep this code which work as-is and we can rework it latter.

Reply:
@ekohl If its OK for you, its ready to be merged.

Reply:
Yes, that feels good. I'm not sure if my code actually **is** the correct solution while yours is "just" moving code around. Generally that's safer. Consider my comment input on a future refactor.